.github/composite-actions/minor-version-upgrade-util/action.yml

```diff
@@ -21,13 +21,24 @@ runs:
   using: "composite"
   steps:    
     - name: Build and run tests for Postgres engine using ${{ inputs.engine_branch }}
+      id: build-modified-postgres-new
+      if: always()
       uses: ./.github/composite-actions/build-modified-postgres
       with:
         engine_branch: ${{ inputs.engine_branch }}
         install_dir: ${{ inputs.install_dir }}
 
+    - name: Compile new ANTLR
+      id: compile-new-antlr
+      if: always() && steps.build-modified-postgres-new.outcome == 'success'
+      uses: ./.github/composite-actions/compile-antlr
+      with:
+        version: ${{ inputs.engine_branch == 'latest' && '4.13.2' || '4.9.3' }}
+        install_dir: ${{inputs.install_dir}}
```

Comment:
Are we building new antlr only for latest version? I think we should instead update this condition to build latest antlr for all the versions from now on.

Reply:
The problem with trying to use the latest antlr for all versions including older versions is that the github workflows will break when trying to compile any older versions. The only way around this would be to find all older versions which are being built from the github actions, and first update them to use antlr-4.13.2. I think that is too much work, which is why I went with this solution of allowing older versions to compile with 4.9.3 and only use 4.13.2 for builds against the latest version.

Reply:
You can see the extremely long chain of commits I had getting this to work, where all of the upgrade tests were failing due to mismatches in antlr version. The main reason why this happens is that all of the github workflows assume that antlr will be compiled separately from the extension, so they were written to compile antlr once before building the older version, and just use that same antlr build for the updated version as well.

Reply:
Building old engine versions with antlr-4.9.3 is fine, what I meant was to build current and newer engine versions with new antlr-4.13.2. For example in future, in version 16.7, the above condition will build only the latest version with antlr-4.13.2 but it will still build 16.6 with old antlr-4.9.3 which doesn't seem right. So, I was thinking if we can make this condition something like:
```
if (version >= 16.6 then use antlr-4.13.2 else use antlr-4.9.3)
```

Reply:
Good point. Let me try to figure out how to set up this condition to work properly.